src/App.jsx

```diff
@@ -0,0 +1,54 @@
+import React from 'react';
+
+import './App.scss';
+import { getAll, get5First, getRed } from './api/goods';
+import { GoodsList } from './components/GoodsList';
+
+class App extends React.Component {
+  state = {
+    goods: [],
+  };
+
+  clickHandler = ({ target }) => {
+    switch (target.name) {
+      case 'first5':
+        get5First()
+          .then(goods => this.setState({ goods }));
+
+        break;
+      case 'red':
+        getRed()
+          .then(goods => this.setState({ goods }));
+
+        break;
+      default:
+        getAll()
+          .then(goods => this.setState({ goods }));
+    }
+  };
```

Comment:
you can simplify this even more and pass to this function a callback (get5First, getRed, getAll), because now you have duplicated code